pkg/solana/chainreader/batch.go

```diff
@@ -31,19 +32,44 @@ type MultipleAccountGetter interface {
 }
 
 func doMethodBatchCall(ctx context.Context, client MultipleAccountGetter, bindings namespaceBindings, batch []call) ([]batchResultWithErr, error) {
-	// Create the list of public keys to fetch
-	keys := make([]solana.PublicKey, len(batch))
+	results := make([]batchResultWithErr, len(batch))
+
+	// create the list of public keys to fetch
+	keys := []solana.PublicKey{}
+
+	// map batch call index to key index (some calls are event reads and will be handled by a different binding)
+	dataMap := make(map[int]int)
+
 	for idx, call := range batch {
 		binding, err := bindings.GetReadBinding(call.ContractName, call.ReadName)
 		if err != nil {
 			return nil, err
 		}
 
+		lvBinding, ok := binding.(latestValueBinding)
+		if ok {
+			results[idx] = batchResultWithErr{
+				address:      "", // TODO: get contract public key
```

Comment:
Just double checking if this TODO is going to be addresses in a follow up PR? Don't we have the contract public key available here through the binding?

Reply:
Fixed